modules/base/scale/timelinescale.cpp

```diff
@@ -0,0 +1,120 @@
+        "keyframes. If this value is set to 'false', the interpolation is not applied.",
+        openspace::properties::Property::Visibility::AdvancedUser
+    };
+
+    // This `Scale` uses a timeline of other `Scale` classes to calculate the final
+    // scale factor for the attached scene graph node. The current in-game time is used to
+    // determine which specific keyframe to currently use. It is also possible to
+    // interpolate between two adjacent keyframes.
+    struct [[codegen::Dictionary(TimelineScale)]] Parameters {
+        // A table of keyframes, with keys formatted as YYYY-MM-DDTHH:MM:SS and values
+        // that are valid Scale objects
+        std::map<std::string, ghoul::Dictionary> keyframes
+            [[codegen::reference("core_transform_scale")]];
+
+        // [[codegen::verbatim(ShouldInterpolateInfo.description)]]
+        std::optional<bool> shouldInterpolate;
+    };
+#include "timelinescale_codegen.cpp"
+} // namespace
+
+namespace openspace {
+
+documentation::Documentation TimelineScale::Documentation() {
+    return codegen::doc<Parameters>("base_transform_scale_keyframe");
+}
+
+TimelineScale::TimelineScale(const ghoul::Dictionary& dictionary)
+    : _shouldInterpolate(ShouldInterpolateInfo, true)
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+
+    for (const std::pair<const std::string, ghoul::Dictionary>& kf : p.keyframes) {
+        const double t = Time::convertTime(kf.first);
+
+        ghoul::mm_unique_ptr<Scale> scale = Scale::createFromDictionary(kf.second);
+        scale->setIdentifier(makeIdentifier(kf.first));
+        addPropertySubOwner(scale.get());
+        _timeline.addKeyframe(t, std::move(scale));
+    }
+
+    _shouldInterpolate = p.shouldInterpolate.value_or(_shouldInterpolate);
+    addProperty(_shouldInterpolate);
+}
+
+glm::dvec3 TimelineScale::scaleValue(const UpdateData& data) const {
+    const double now = data.time.j2000Seconds();
+    using KeyframePointer = const Keyframe<ghoul::mm_unique_ptr<Scale>>*;
+
+    KeyframePointer prev = _timeline.lastKeyframeBefore(now, true);
+    KeyframePointer next = _timeline.firstKeyframeAfter(now, true);
+
+    if (!prev && !next) {
+        return glm::dvec3(0.0);
+    }
+    if (!prev) {
+        prev = next;
+    }
+    if (!next) {
+        next = prev;
+    }
+    const double prevTime = prev->timestamp;
+    const double nextTime = next->timestamp;
+
+    if (_shouldInterpolate) {
+        double t = 0.0;
+        if (nextTime - prevTime > 0.0) {
+            t = (now - prevTime) / (nextTime - prevTime);
+        }
+        return glm::mix(prev->data->scaleValue(data), next->data->scaleValue(data), t);
+    }
+    else {
+        if (prevTime <= now && now < nextTime) {
+            return prev->data->scaleValue(data);
+        }
+        else if (nextTime <= now) {
+            return next->data->scaleValue(data);
+        }
```

Comment:
I think we will need a case for if `now` is before `prevTime` somewhere here. When I try the non-interpolating example, it disappears when the current time is set to before the first keyframe. This is not the case when I use the interpolating example.